workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -125,7 +185,52 @@ export class DynamicAuthorizationService {
    *
    */
   public async isAuthorizedOnRoute(isAuthorizedOnRouteRequest: IsAuthorizedOnRouteRequest): Promise<void> {
-    throw new Error('Not implemented');
+    const validatedRequest = IsAuthorizedOnRouteRequestParser.parse(isAuthorizedOnRouteRequest);
+    const { authenticatedUser, route, method, params } = validatedRequest;
+    const metadata: Metadata = {
+      actor: authenticatedUser,
+      action: this.isAuthorizedOnRoute.name,
+      source: {
+        serviceName: DynamicAuthorizationService.name
+      },
+      requestBody: validatedRequest
+    };
+    try {
+      if ((await this.isRouteIgnored({ route, method })).data.routeIgnored) {
+        metadata.statusCode = 200;
+        await this._auditService.write(metadata);
+        return;
+      }
+      if (!(await this.isRouteProtected({ route, method })).data.routeProtected)
+        throw new RouteNotSecuredError('Route is not secured');
+      const { data: dynamicOperationsData } =
+        await this._dynamicAuthorizationPermissionsPlugin.getDynamicOperationsByRoute({ route, method });
+      const dynamicOperations = _.cloneDeep(dynamicOperationsData.dynamicOperations);
+      // Inject subject's variable based params
+      const paramRegex = /\${([^{]+)}/g;
```

Comment:
Is the data being check by this regex used / stored in a way that would allow malicious input to get stored / executed?

Reply:
regex is a safe regex using [safe-regex](https://www.npmjs.com/package/safe-regex) to detect if it can cause exponential time regex evaluation. This should prevent a ReDos attack. The IsAuthorizedOnRouteRequestParser checks to ensure the values being injected are strings. These are later evaluated by CASL which should sanitize any form of string execution

Reply:
I might be missing something, but if the only requirement here is to make sure this is non-executable string, I'd recommend escaping the string value rather than trying to parse it out with a regex. That should prevent all sorts of potential malicious input attacks.

Reply:
resolved by escaping user input